### PR TITLE
Fix /mode command to correctly assume target

### LINKED ISSUE #676: Cannot send multiple mode changes at once
Sending `/mode #channel -ao nick nick` only actually performs the -a. This implies that The Lounge intercepts mode commands and doesn't pass them to the server verbatim.

Came across this while running tests for #675.

### PR DESCRIPTION
**Pull request by xPaw**
Fixes #676 and #223.

Also adds support for passing in multiple nicks into `/op` (and other short commands). Adds a `/hop` and `/dehop` aliases.

### CODE REVIEW

File: src/plugins/inputs/mode.js [hunk 1]
```diff
@@ -1,29 +1,57 @@
...
+	if (args.length === 0 || args[0][0] === "+" || args[0][0] === "-") {
+		args.unshift(chan.type === Chan.Type.CHANNEL || chan.type === Chan.Type.QUERY ? chan.name : network.nick);
 	}
 
-	var irc = network.irc;
-	irc.raw("MODE", chan.name, mode, user);
+	args.unshift("MODE");
 
-	return true;
+	network.irc.raw.apply(network.irc, args);
```
Comment (xPaw) [inline src/plugins/inputs/mode.js]: `...` spread operator is not supported in node4.

File: src/plugins/inputs/mode.js [hunk 2]
```diff
@@ -1,29 +1,57 @@
...
+
+		args.forEach(function(target) {
+			network.irc.raw("MODE", chan.name, mode, target);
+		});
+
+		return;
+	}
+
+	if (args.length === 0 || args[0][0] === "+" || args[0][0] === "-") {
+		args.unshift(chan.type === Chan.Type.CHANNEL || chan.type === Chan.Type.QUERY ? chan.name : network.nick);
```
Comment (xPaw) [inline src/plugins/inputs/mode.js]: Not sure if there's any use of allowing `query` targets here.

File: test/commands/mode.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+			name: "Network Lobby",
+			type: Chan.Type.LOBBY
+		});
+
+		const testableNetwork = {
+			lastCommand: null,
+			nick: "xPaw",
+			irc: {
+				raw: function() {
+					testableNetwork.lastCommand = Array.prototype.join.call(arguments, " ");
```
Comment (astorije) [inline test/commands/mode.js]: Nice trick to get the latest command called!

File: test/commands/mode.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			ModeCommand.input(testableNetwork, channel, "voice", ["xPaw"]);
+			expect(testableNetwork.lastCommand).to.equal("MODE #thelounge +v xPaw");
+
+			ModeCommand.input(testableNetwork, channel, "devoice", ["xPaw"]);
+			expect(testableNetwork.lastCommand).to.equal("MODE #thelounge -v xPaw");
+
+			// Multiple arguments are supported, sent as separate commands
+			ModeCommand.input(testableNetwork, channel, "devoice", ["xPaw", "Max-P"]);
+			expect(testableNetwork.lastCommand).to.equal("MODE #thelounge -v Max-P");
+		});
```
Comment (astorije) [inline test/commands/mode.js]: I'd rather see these broken up in multiple expects (can be grouped in a common `describe`), but really no big deal, especially at that point.
Awesome job on adding tests though!!

**Pull request merged**